src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +23,41 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+
+        if (obj.getClass() != this.getClass()) {
+            return false;
+        }
+
+        Rectangle rectangle = (Rectangle) obj;
+        if ((width == rectangle.width) && (length != rectangle.length)
+                && (color.equals(rectangle.getColor()))) {
```

Comment:
so it can return true even if length doesn't equal?

Reply:
what if color would be null?

Reply:
done